EIPS/eip-2315.md

```diff
@@ -95,43 +105,42 @@ We would like to consider EVM code valid iff no execution of the program can lea
+* prevents stack underflow
+* ensures that all calls to a subroutine have the same number of inputs and the same number of outputs and
+* ensures that stack height is bounded in the absence of recursion.
 
-Requiring a consistently aligned`data stack` also
-* prevents stack underflow,
-* ensures that all calls to a subroutine have the same number of inputs and the same number of outputs,
-* ensures that stack height is bounded, (in the absence of recursion) and
-* ensures that code validation and compilation algorithms that traverse the control-flow graph can break cycles at joins, which allows us to implement streaming, one-pass, and other quasi-linear compilers.
+Requiring a consistently aligned`data stack` also allows some algorithms that traverse the control-flow graph -- includinng code validation and compilation -- to break cycles at joins, again preventing quadratic path explosion.  When a traversal gets to a `PC` it has visited before it is either at the beginning of a loop or the entry to a function.  Since the stacks align we know that loops will not grow, or that the number of arguments is the same, so there is no need to continue that path.
 
-_Note: The JVM and Wasm enforce similar constraints on the stack for similar reasons._
+_Note: The JVM and Wasm enforce similar constraints for similar reasons._
 
 ### Alternative Designs
 
-There are a few major designs for a subroutine facility, two of which are considered here.  The others are mostly not appropriate for the EVM, such as self-modifying code that writes return addresses into called subroutines.
+There are a few major designs for a subroutine facility, two of which are considered here.  The others are mostly not appropriate for the EVM, such as the Wheeler Jump -- self-modifying code that writes return addresses into called subroutines.
 
 *1. Keep return addresses on a dedicated return stack.* Turing's design is often used by stack machines, including those for Forth, Java, Wasm, and others.  The data stack is used almost entirely for computation, with a dedicated stack for return addresses.  A single instruction suffices to call, and another to return from a routine.
 
-*2. Keep return addresses on the data stack.*  This design is often used by register machines, including CDC, IBM, DEC, Intel, and ARM.  The registers are used primarily for computation, and the stack maintains call frames for return addresses, arguments, and local variables.  On the EVM there are no registers for computation, so using the stack for both purposes can be less than ideal.
+*2. Keep return addresses on the data stack.*  This design is often used by register machines, including those from CDC, IBM, DEC, Intel, and ARM.  The registers are used primarily for computation, and the stack maintains call frames for return addresses, arguments, and local variables.  On the EVM there are no registers for computation, so using the stack for both purposes can be less than ideal.
```

Comment:
In earlier versions I referenced only some of the models I had actually programmed, (Burroughs 5000, CDC 7600, IBM 360, DEC PDP-11 and VAX, M68000, Sun SPARC, ARM, and Intel x86s) having been accused of "arguing from authority."  I've since decided that I am an authority, and that for a "person skilled in the art" I didn't need to be that specific.

Reply:
I obviously held different views with whoever made that accusation. My stance is that: so long as the EIP content is not being in-appropriately promoting, the more specific and well-referencing, the better. It helps readability.

Please feel free to ignore or make your own judgement (as the author EIP). And I'd argument _for_ author judgement.

Reply:
I'm walking a Iine between keeping it brief and keeping it informative, thanks.  As the pure spec that @MicahZoltu would prefer (and I would to, in my way) I could just add a few lines to the abstract and stop there.

Reply:
My stance is 
1. for Spec be as brief as possible, but
2. for Motivation and Rationale, be as informative as possible. 

Since this is in the Rationale / Alternative Design, I think it's ok to be informative.
Going back to what "EIP" means to me, EIP spells Ethereum Improvement Proposal. Therefore, as a "proposal", I think it's good to elaborate reasoning as long as it's not duplicative and repetitive.